### DISCUSSION TITLE
Random state in plot_gain

### DISCUSSION THREAD

**Discussion by AlexanderTrg** [uber/causalml#751] [Q&A]
Hey, can you help me with clarification of how random state works? 
1. when plot_gain chart is drawn, the Y value changes when the random stat is changed.
2. when setting bin = 2, the random state has no effect on the line. The start and end are not changed.
3. when bin = 2, the cumulative effect is calculated as (mean score treatment =1 - mean score treatment =0) * sample size. How does it count for calculating Y when bin = 3 (for point X equal to half the sample size)?

![image](https://github.com/uber/causalml/assets/37145876/47d24c59-d497-48ec-a5b9-c78643d2e72c)
![image](https://github.com/uber/causalml/assets/37145876/35405398-4c8e-4e6f-9e11-2d521e78bc40)

**Comment by jeongyoonlee** [comment]
Can you share the code you used to generate the plots? `plot_gain()` doesn't take `bin` as an input argument ([ref](https://github.com/uber/causalml/blob/a0315660d9b14f5d943aa688d8242eb621d2ba76/causalml/metrics/visualize.py#L504)).

**Reply by AlexanderTrg** [reply]
> Can you share the code you used to generate the plots? `plot_gain()` doesn't take `bin` as an input argument ([ref](https://github.com/uber/causalml/blob/a0315660d9b14f5d943aa688d8242eb621d2ba76/causalml/metrics/visualize.py#L504)).

n is bin. isnt? 
n (int, optional): the number of samples to be used for plotting
